File: Animator/C#/PatternsAnimator/Lib/CompositeEffect.cs
using System.Collections.Generic;

namespace PatternsAnimator.Lib
{
    public class CompositeEffect<T> : EffectBase<T>
    {
        private readonly List<IEffect<T>> _parts;

        public CompositeEffect()
        {
            _parts = new List<IEffect<T>>();
        }

        public void AddPart(IEffect<T> e)
        {
            _parts.Add(e);
        }

        public override void Animate(T o)
        {
            foreach (var part in _parts)
            {
                part.Animate(o);
            }
        }

        protected override void DoAnimation(T o)
        {
        }
    }
}

File: Animator/Java/src/Lib/CompositeEffect.java
package Lib;

import java.util.ArrayList;
import java.util.List;

public class CompositeEffect<T> extends EffectBase<T> {
    private final List<IEffect<T>> parts;

    public CompositeEffect() {
        this.parts = new ArrayList<>();
    }

    public void addPart(IEffect<T> e) {
        this.parts.add(e);
    }

    @Override
    public void animate(T o) {
        for (var p : this.parts) {
            p.animate(o);
        }
    }

    @Override
    protected void doAnimation(T o) {
    }
}
